library/src/c_api/mod.rs

```diff
@@ -46,6 +50,15 @@ pub struct AppParameters {
     pub code_cache_dir: *const libc::c_char,
 }
 
+#[derive(Clone, Copy, Debug)]
+#[repr(C)]
+pub struct FileCallbacks {
+    pub open: extern "C" fn() -> *mut libc::c_void,
```

Comment:
I'd give them argument names and comments for your own later sanity, but that can also be a follow-up.

Reply:
Done